### PR TITLE
Docs: Add mentee selection process

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR adds content for the `Mentee Selection` section i.e, process and recommendations,  based on recent mentor learnings meeting we had with @bwplotka, @fpetkovski & @metonymic-smokey.

Would love to get some feedback on these processes and any additional recommendations that you might want! 🙂 

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: docs/contributing/mentorship.md [hunk 1]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
```
Comment (matej-g) [inline docs/contributing/mentorship.md:123]: I'd even say debugging local environment is not strictly part of mentoring, ideally this should be up to the mentee and they should be able to deal with their environment (obviously, mentor should be able to follow their environment during pair programming etc.).

File: docs/contributing/mentorship.md [hunk 2]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
```
Comment (matej-g) [inline docs/contributing/mentorship.md:120]: This sounds like condition sine qua non, do we want to have it explicit here?

File: docs/contributing/mentorship.md [hunk 3]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
+    * The mentee should have basic git, bash, programming, and Go knowledge. 
+    * They should be using some common code editor/IDE (VSCode/Goland), for the duration of mentorship, so that it is easier to pair program and share knowledge with them, without being blocked by things like code navigation.
```
Comment (matej-g) [inline docs/contributing/mentorship.md:125]: See above, I would say leave this up to the mentee, but it should not be _too_ exotic. But e.g. if they want to use Emacs we should not be stopping them :joy:.

File: docs/contributing/mentorship.md [hunk 4]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+  * It is preferable to have mentees from historically underrepresented communities in tech, to ensure diverse contributors in Thanos.
```
Comment (matej-g) [inline docs/contributing/mentorship.md:126]: :+1:

**Review by matej-g — COMMENTED** [review]
Nice to have this in writing :blush:

File: docs/contributing/mentorship.md (hunk 1 above)
Comment (squat) [reply docs/contributing/mentorship.md:123]: I don't think this is strictly necessary and I think it's potentially exclusionary to note it. If mentees want to use windows because it's the machine they have access to, then by all means, let them. They can always use the WSL or something. Alternatively, they can help get our scripts and such working on .ore platforms.

File: docs/contributing/mentorship.md [hunk 5]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
```
Comment (squat) [inline docs/contributing/mentorship.md:116]: This is neither required nor expected. We can note that having prior contributions is favorable because it demonstrates willingness to engage with the community and independence.

File: docs/contributing/mentorship.md [hunk 6]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
```
Comment (squat) [inline docs/contributing/mentorship.md:119]: Let's avoid the potentially judgmental descriptor "proper".
```suggestion
  * The mentee should have access to a stable internet connection over which they can attend regular video calls with mentors.
```

File: docs/contributing/mentorship.md [hunk 7]
```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
...
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
+    * The mentee should have basic git, bash, programming, and Go knowledge.
```
Comment (squat) [inline docs/contributing/mentorship.md:124]: This is confusing. Mentees are not expected to necessarily know Go. There are many frontend student projects as well as potential projects centered around documentation or other community work.

File: docs/contributing/mentorship.md (hunk 3 above)
Comment (squat) [inline docs/contributing/mentorship.md:125]: This is odd to me. I'm a maintainer and a mentor and I've never used vscode or goland and would not expect any mentee to be required to do so, especially when the tools (like goland) are closed-source and paid.

Comment (saswatamcode) [reply docs/contributing/mentorship.md:125]: So the main motivation for putting this here is to ensure that the mentee uses some editor/IDE that allows them to navigate code and lookup definitions easily across large codebases. We've had some experiences where mentees weren't progressing or getting frequently blocked due to using a very simplistic text editor or something so complex, that they don't know how to operate it well. So they get confused when trying to think of solutions, and have to ask code-level questions frequently. This can be avoided if mentors already know about this while interviewing, and can suggest they switch to something easier, to have a more productive mentorship!

That being said, let me rephrase this line to capture this, and not make it seem like vscode/goland are the only options! 🙂

File: docs/contributing/mentorship.md (hunk 7 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:124]: Will remove!

File: docs/contributing/mentorship.md (hunk 5 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:116]: Let me rephrase here.

File: docs/contributing/mentorship.md (hunk 3 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:125]: Yup, will rephrase as mentioned here: https://github.com/thanos-io/thanos/pull/5920#discussion_r1032731115

File: docs/contributing/mentorship.md (hunk 2 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md]: So we didn't interview before, and this condition wasn't being checked, which led to some troublesome experiences. It is sine qua non, but mentioning it explicitly here, so that mentors can be mindful of this as a prerequisite. WDYT? 🙂

File: docs/contributing/mentorship.md (hunk 1 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:123]: Yup, I can rephrase a bit here but generally wanted to mention that having a local environment that the mentee knows how to operate, is beneficial, and helps the mentorship be more productive. I think we can mention linux VM as a fallback if problems due to this arise. We're not excluding any platforms/machines by any means. WDYT? 🙂 

I think @bwplotka can speak more to this as well.

File: docs/contributing/mentorship.md (hunk 3 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md]: Have updated this!

File: docs/contributing/mentorship.md [hunk 8]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply docs/contributing/mentorship.md:115]: Maybe it's too intrusive, but what if we ask for some proof, forwarded email or signed letter from manager that they are allowed to work on mentorship? (:

File: docs/contributing/mentorship.md [hunk 9]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+  * It is preferable for the mentee to have at least one prior contribution to Thanos or its dependencies that affect it indirectly. Such prior contributions demonstrate a willingness to engage with the community and independence from the potential mentee.
+    * However, the contribution should not be a minimal effort one, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have access to a stable internet connection over which they can attend regular video calls with mentors (can be ensured during a 20-minute interview).
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
```
Comment (bwplotka) [inline docs/contributing/mentorship.md:119]:
```suggestion
    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well (verbally and in written) in English (first/second language).
```

File: docs/contributing/mentorship.md (hunk 4 above)
Comment (bwplotka) [reply docs/contributing/mentorship.md:124]: Can we put this as the first item? Quite important.

**Review by bwplotka — APPROVED** [review]
Nice, thanks!

File: docs/contributing/mentorship.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (saswatamcode) [reply docs/contributing/mentorship.md]: Added line about it. 🙂

File: docs/contributing/mentorship.md (hunk 4 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:112]: Moved to top

File: docs/contributing/mentorship.md [hunk 10]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they claim to commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and not being able to utilize the time mentors can spend with them. It is hard to help them in such case. So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally, we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution. Generally, it's only ok to have a full-time job only for more senior engineers, who understand how to organize work, be transparent about mentorship with their employer etc.
+      * In case the mentee is employed full-time, asking for some form of proof (letter/forwarded email from manager) that their employer approves of them working on this mentorship is preferable.
```
Comment (bwplotka) [inline docs/contributing/mentorship.md:116]:
```suggestion
      * In case the mentee is employed full-time, we recommend asking for some form of proof (letter/forwarded email from manager) that their employer approves of them working on this mentorship is preferable.
```

**Review by bwplotka — DISMISSED** [review]
Thanks! Great, we can always iterate later. @squat are you roughly ok with wording here?

File: docs/contributing/mentorship.md (hunk 10 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:116]: "is preferable" is at the end, so that makes this optional anyway 🙂

File: docs/contributing/mentorship.md [hunk 11]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
 
 ### Mentees Selection
 
-TBD
+This section is to help guide mentors, old and new, on how to select mentees for Thanos mentorship projects. It details recommendations and steps for this entire process.
+
+#### Process
+
+The steps for any particular project mentee selection are as follows:
```
Comment (squat) [inline docs/contributing/mentorship.md:89]:
```suggestion
We follow these steps when selecting mentees for any given project:
```

File: docs/contributing/mentorship.md [hunk 12]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+
+1. After your project has been proposed to a particular mentorship program like LFX or GSoC, ensure that it is discoverable by potential mentees during the application period, by sharing it on the relevant GitHub issue. You could also optionally share this on the [#thanos-dev slack channel](https://cloud-native.slack.com/archives/CL25937SP), [Twitter](https://twitter.com/ThanosMetrics), and even during Thanos Contributor Hours.
```
Comment (squat) [inline docs/contributing/mentorship.md:91]: Unnecessary comma 
```suggestion
1. After your project has been proposed to a particular mentorship program like LFX or GSoC, ensure that it is discoverable by potential mentees during the application period by sharing it on the relevant GitHub issue. You could also optionally share this on the [#thanos-dev slack channel](https://cloud-native.slack.com/archives/CL25937SP), [Twitter](https://twitter.com/ThanosMetrics), and even during Thanos Contributor Hours.
```

File: docs/contributing/mentorship.md [hunk 13]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+2. Wait for the mentee application period to end, and log into the relevant portal for the mentorship, in order to view mentee applications. You can refer to the relevant documentation that you will receive, to familiarize yourself with the portal, application format, and **selection deadlines**. Additional reference: [LFX Mentor docs](https://docs.linuxfoundation.org/lfx/mentorship/mentors/review-mentee-applications-and-tasks), [GSoC mentor guide](https://google.github.io/gsocguides/mentor/).
```
Comment (squat) [inline docs/contributing/mentorship.md:92]: unnecessary commas
```suggestion
2. Wait for the mentee application period to end and log into the relevant portal for the mentorship in order to view mentee applications. You can refer to the relevant documentation that you will receive to familiarize yourself with the portal, application format, and **selection deadlines**. Additional reference: [LFX Mentor docs](https://docs.linuxfoundation.org/lfx/mentorship/mentors/review-mentee-applications-and-tasks), [GSoC mentor guide](https://google.github.io/gsocguides/mentor/).
```

File: docs/contributing/mentorship.md [hunk 14]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+3. Discuss with your co-mentors, and shortlist roughly, 30% of the mentee applications that you've found to be suitable, by referring to their applications, which can consist of things like CVs, proposals, and "statement of purpose" type questions. Refer to [recommendations](#recommendations) for what to keep in mind while shortlisting.
```
Comment (squat) [inline docs/contributing/mentorship.md:93]: commas and double 'application'
```suggestion
3. Discuss with your co-mentors and shortlist roughly 30% of the mentees that you've found to be suitable by referring to their applications, which can consist of things like CVs, proposals, and "statement of purpose" type questions. Refer to [recommendations](#recommendations) for what to keep in mind while shortlisting.
```

File: docs/contributing/mentorship.md [hunk 15]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+4. Send an email with a few extra questions to get to know these 30% potential mentees even more. You can refer to the `thanos-io@googlegroups.com` mailing list for a list of such questions and the email format from past mentorships. Make sure to add mentee emails to `bcc` and your co-mentors and `thanos-io@googlegroups.com` to `to`. Set a deadline for the responses.
+5. Once you receive replies from most of them, review again with your co-mentor and shortlist candidates that you would now like to interview.
+6. Conduct 20-minute interviews (refer to [recommendations](#recommendations) for what to keep in mind during these interviews) of the shortlisted mentees to get to know them even better, and finally select a single person.
```
Comment (squat) [inline docs/contributing/mentorship.md:96]:
```suggestion
6. Conduct 20-minute interviews (refer to [recommendations](#recommendations) for what to keep in mind during these interviews) of the shortlisted mentees to get to know them even better and finally select a single person.
```

File: docs/contributing/mentorship.md [hunk 16]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+7. Send an email to the selected mentee asking for confirmation, and if they are happy with this.
```
Comment (squat) [inline docs/contributing/mentorship.md:97]:
```suggestion
7. Send an email to the selected mentee asking for confirmation and if they are happy with this.
```

File: docs/contributing/mentorship.md [hunk 17]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+8. Once they reply, fill out the selection form in the relevant portal! :)
+
+#### Recommendations
+
+During mentee selection for our various mentorship programs, we want to follow some best practices to ensure that the mentorship is a fruitful experience for both mentees and mentors. A large part of ensuring this is selecting the most suitable candidate for the mentorship!
+
+This process is hard and requires time, as it involves holistically looking at mentorship applications and trying to ascertain if they would be able to benefit from this mentorship and ideally become a contributor, maintainer or hire in near future.
```
Comment (squat) [inline docs/contributing/mentorship.md:104]: > hire in the near future.

hire by whom? Thanos isn't a company

File: docs/contributing/mentorship.md [hunk 18]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+
+With several past mentorships, we have been able to distil some learnings from the process, which are listed here as recommendations to mentors:
```
Comment (squat) [inline docs/contributing/mentorship.md:106]:
```suggestion
We have been able to distill some learnings from several past mentorship programs, which are listed here as recommendations to mentors:
```

File: docs/contributing/mentorship.md [hunk 19]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+  * It is preferable to have mentees from historically underrepresented communities in tech, to ensure diverse contributors in Thanos.
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they claim to commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and not being able to utilize the time mentors can spend with them. It is hard to help them in such case. So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally, we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution. Generally, it's only ok to have a full-time job only for more senior engineers, who understand how to organize work, be transparent about mentorship with their employer etc.
```
Comment (squat) [inline docs/contributing/mentorship.md:115]:
```suggestion
    * Generally, we expect our mentees to be university/college students who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution. Generally, it's only ok to have a full-time job only for more senior engineerswho underst and how to organize their work, be transparent about the mentorship with their employer, etc.
```

File: docs/contributing/mentorship.md [hunk 20]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+      * In case the mentee is employed full-time, asking for some form of proof (letter/forwarded email from manager) that their employer approves of them working on this mentorship is preferable.
+  * It is preferable for the mentee to have at least one prior contribution to Thanos or its dependencies that affect it indirectly. Such prior contributions demonstrate a willingness to engage with the community and independence from the potential mentee.
+    * However, the contribution should not be a minimal effort one, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
```
Comment (squat) [inline docs/contributing/mentorship.md:118]:
```suggestion
    * However, the contribution should not be a minimal effort one, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort. A better indicator would be some change to the code or documentation. Note that this contribution doesn't need to be in a merged or finalized state and can be open.
```

File: docs/contributing/mentorship.md [hunk 21]
```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
...
+  * The mentee should have access to a stable internet connection over which they can attend regular video calls with mentors (can be ensured during a 20-minute interview).
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well (verbally and in written) in English (first/second language).
+  * The mentee should have some form of local development environment setup that they can operate which would not require the mentor to constantly debug the mentee's environment while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS/, so most generic GNU/BSD commands are compatible.
+    * The mentee should have basic git, bash, and programming knowledge.
+    * They should be using some common code editor/IDE (VSCode/Goland etc) that they can operate comfortably, for the duration of mentorship. It should allow them to easily navigate code and lookup definitions across large codebases as this makes it easier to pair program and share knowledge with them.
```
Comment (squat) [inline docs/contributing/mentorship.md:124]:
```suggestion
    * They should be using a common code editor with IDE functionality for Golang, e.g. VSCode, Goland etc, that they can operate comfortably for the duration of mentorship. The editor should allow them to easily navigate code and lookup definitions across large codebases as this makes it easier to pair program and share knowledge with them.
```

**Review by squat — COMMENTED** [review]
generally looks good. made some more suggestions. phrasing looks pretty good

File: docs/contributing/mentorship.md (hunk 1 above)
Comment (douglascamata) [reply docs/contributing/mentorship.md]: > Unfortunately, we have a visible case where WSL blocked mentees for the majority of mentorship time.

Let's make this better then. I have a WSL2 environment at home that I successfully used for some time to work on Thanos and if there are any related issues I could have a look at them to help.

I 200% agree with @squat's statement.

File: docs/contributing/mentorship.md [hunk 22]
```diff
@@ -63,7 +63,7 @@ In order to allow mentees to share the knowledge their learned and allow them to
 
 ## For Mentors
 
-This section explains some process around selecting mentees (TBD) and [retrospective](#retrospective) process.
+This section explains some process around selecting mentees and [retrospective](#retrospective) process.
```
[the remark this answers is not in the document]
Comment (saswatamcode) [reply docs/contributing/mentorship.md:66]:
```suggestion
This section explains processes around selecting mentees and conducting [retrospectives](#retrospective).
```

File: docs/contributing/mentorship.md (hunk 17 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md:104]:
```suggestion
This process is hard and requires time, as it involves holistically looking at mentorship applications and trying to ascertain if they would be able to benefit from this mentorship and ideally become a contributor, maintainer, or even new hire into the respective mentor's company in near future.
```

File: docs/contributing/mentorship.md (hunk 1 above)
Comment (saswatamcode) [reply docs/contributing/mentorship.md]: To reiterate, we aren't excluding mentees from using any machine/OS. Running a linux VM is acceptable in most cases and this is about having really productive mentorship, without the mentor having to spend time, debugging the mentee's local env in case they run into problems and are blocked. 

> Let's make this better then. I have a WSL2 environment at home that I successfully used for some time to work on Thanos and if there are any related issues I could have a look at them to help.

You could update https://thanos.io/tip/contributing/contributing.md/#development with this, but I'd still be against recommending complicated envs to mentees (who are often new to software and open-source), and just defaulting to simple linux VMs.

**Review by squat — DISMISSED** [review]
lgtm

File: docs/contributing/mentorship.md (hunk 4 above)
Comment (fpetkovski) [reply docs/contributing/mentorship.md:112]: Personally I would not put too much emphasis on this and make it a nice to have. Otherwise we are discriminating against people who are a good fit for various other reasons. I would not give this more weight than any other factor, and treat it as more of a tie-breaker.

**Review by bwplotka — APPROVED** [review]

File: docs/contributing/mentorship.md (hunk 2 above)
Comment (matej-g) [reply docs/contributing/mentorship.md]: All good, if there's a rationale for it as you said, then it should be here :+1:

File: docs/contributing/mentorship.md (hunk 1 above)
Comment (douglascamata) [reply docs/contributing/mentorship.md]: It's not complicated, unless we have Service objects using `service.spec.sessionAffinity: ClientIP`. In this case we would need to recompile the WSL 2 kernel, then it becomes complicated.

Otherwise it's literally a matter of installing an app from the Microsoft Store. It integrates easily into Windows itself and any GUI code editor (like, but not limited to, VSCode or Goland).
